File: BackEnd/apiFunctions/assembly.php
<?php
//*************************************************************************************************
// FileName : assembly.php
// FilePath : apiFunctions/
// Author   : Christian Marty
// Date		: 21.11.2023
// License  : MIT
// Website  : www.christian-marty.ch
//*************************************************************************************************
declare(strict_types=1);
global $database;
global $api;

if($api->isGet(Permission::Assembly_List))
{
    $query = <<< QUERY
        SELECT 
            AssemblyNumber,
            assembly.Name AS Name,
            assembly.Description AS Description,
            productionPart.Number AS ProductionPartNumber,
            numbering.Prefix AS ProductionPartNumberPrefix
        FROM assembly
        LEFT JOIN productionPart ON assembly.ProductionPartId = productionPart.Id
        LEFT JOIN numbering ON productionPart.NumberingPrefixId = numbering.Id
    QUERY;
    $result = $database->query($query);
    \Error\checkErrorAndExit($result);

	foreach($result as $item) {
        $item->AssemblyNumber = intval($item->AssemblyNumber);
        $item->ItemCode =  \Numbering\format(\Numbering\Category::Assembly, $item->AssemblyNumber);
        if($item->ProductionPartNumber !== null){
            $item->ProductionPartCode = \Numbering\format(\Numbering\Category::ProductionPart, $item->ProductionPartNumberPrefix."-".$item->ProductionPartNumber);
        }else{
            $item->ProductionPartCode = null;
        }
        unset($item->ProductionPartNumber);
        unset($item->ProductionPartNumberPrefix);
	}
    $api->returnData($result);
}

File: API/assembly.py
# *************************************************************************************************
# FileName : assembly.py
# Author   : Christian Marty
# Date     : 16.02.2024
# License  : MIT
# Website  : www.christian-marty.ch
# *************************************************************************************************

class AssemblyUnitHistory(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance

    def item(self, assembly_unit_history_code: str | int):
        data = self._eerp.get('assembly/unit/history/item', {"AssemblyUnitHistoryNumber": assembly_unit_history_code})
        return data

    def type(self):
        data = self._eerp.get('assembly/unit/history/type')
        return data


class AssemblyUnit(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance
        self.history = AssemblyUnitHistory(eerp_instance)

    def item(self, assembly_unit_code: str | int):
        data = self._eerp.get('assembly/unit/item', {"AssemblyUnitNumber": assembly_unit_code})
        return data


class Assembly(object):
    def __init__(self, eerp_instance):
        self._eerp = eerp_instance
        self.unit = AssemblyUnit(eerp_instance)

    def list(self):
        data = self._eerp.get('assembly')
        return data

    def item(self, assembly_code: str | int):
        data = self._eerp.get('assembly/item', {"AssemblyNumber": assembly_code})
        return data
